ql/src/java/org/apache/hadoop/hive/ql/parse/HiveTableName.java

```diff
@@ -38,37 +38,22 @@ public HiveTableName(String catName, String dbName, String tableName) {
    * @throws SemanticException
    */
   public static TableName of(Table table) throws SemanticException {
-    return ofNullable(table.getTableName(), table.getDbName());
+    return ofNullable(table.getTableName(), table.getDbName()); // todo: this shouldn't call nullable
   }
 
   /**
-   * Set a @{@link Table} object's table and db names based on the provided string.
-   * @param dbTable the dbtable string
+   * Set a @{@link Table} object's table and db names based on the provided tableName object.
+   * @param tableName the tableName object
    * @param table the table to update
    * @return the table
    * @throws SemanticException
    */
-  public static Table setFrom(String dbTable, Table table) throws SemanticException{
-    TableName name = ofNullable(dbTable);
-    table.setTableName(name.getTable());
-    table.setDbName(name.getDb());
+  public static Table setFrom(TableName tableName, Table table) throws SemanticException{
```

Comment:
this would be better served as an instance method - I guess it can't be added to Table...
how abut something like `tableName.writeInto(table)`

Reply:
I went with this because Table is in ql.metadata, while TableName is in storage-api.